admin/services/check-form.service.js

```diff
@@ -126,6 +126,32 @@ const checkFormService = {
       const promises = modifiedCheckWindows.map(cw => { cw.save() })
       return Promise.all(promises)
     }
+  },
+  /**
+   * Append check windows name(s) and canDelete to check form object.
+   * @param checkWindows
+   * @returns {Promise.<void>}
+   */
+  checkWindowNames: (checkWindows) => {
+    let checkWindowsName = []
+    checkWindows.forEach(cw => {
+      checkWindowsName.push(' ' + cw.checkWindowName)
+    })
+    return checkWindowsName
+  },
+  /**
+   * Append check windows name(s) and canDelete to check form object.
+   * @param checkWindows
+   * @returns {Promise.<void>}
+   */
+  canDelete: (checkWindows) => {
+    let canDelete
```

Comment:
I'd prefer to default to false, and change it to true on the basis of a check.  This looks like it might be returning, false, or undefined.

Also it does not return a Promise, rather a boolean